### ISSUE TITLE
user behaviour on notification of risk

### ISSUE THREAD

**Issue by pylls** [DP-3T/documents#22]
> If the computed risk score is above the provided threshold, the app shows a notification to the user that she/he has been in proximity to an infected patient. The notification contains instructions on what to do and whom to contact.
...
> Hence, from the perspective of an outside observer, the phones of both at-risk persons, those who have been in contact with an infected person, and those who are not at risk, behave the same.

This depends on what is the expected user behaviour on notification of risk and if it's possible to observe contacts made as per the instructions. An attacker able to assert control over the notification service (e.g., by simply dropping traffic) can selectively notify users and observe their behaviour. Reasonable tweaks:

- randomize when the risk score is computed locally instead of doing it immediately upon receiving infected EphIDs (less granularity for such as an attacker)
- recommend that the notification instructions point to some commonly used means of contacting health services, such as through the regional/national healthcare portal, instead of a specific means for infected patients (less distinct network traffic to observe)

The randomization has the added benefit of distributing the load on the contact point.

**Comment by lbarman** [comment]
Hi @pylls , thanks for your input.
> An attacker able to assert control over the notification service (e.g., by simply dropping traffic) can selectively notify users and observe their behaviour. 

It can drop messages, but cannot easily generate at-risk events.  In addition, it could selectively drop messages, but it cannot guess which users are going to raise an at-risk alert (as this computation is done locally from public data). Its power is therefore fairly limited (to DOS'ing the system - which is another problem).

Your second proposition is very valid; it is not clear yet if the application directly shows "you're at risk" or advise to contact health authorities, this will need to be designed with health professionals.

Would this answer your question ? Thanks

**Comment by pylls** [comment]
> Hi @pylls , thanks for your input.

Thanks for taking the time to reply and your work!

> > An attacker able to assert control over the notification service (e.g., by simply dropping traffic) can selectively notify users and observe their behaviour.
> 
> It can drop messages, but cannot easily generate at-risk events. In addition, it could selectively drop messages, but it cannot guess which users are going to raise an at-risk alert (as this computation is done locally from public data). Its power is therefore fairly limited (to DOS'ing the system - which is another problem).

If "but it cannot guess which users are going to raise an at-risk alert " is true or not all depends on what happens when an at-risk alert is raised. This goes back to the analysis I quoted above, "an outside observer, the phones of both at-risk persons, those who have been in contact with an infected person, and those who are not at risk, behave the same" in the same way. You cannot make this statement without taking the vital part of the system of what happens on an at-risk alert into account. 

Imagine the flawed design of an alert that points people to https://example.com/learn-more-about-staying-safe-at-home, a website not regulary visited by people other than for the purpose of this alert (same argument made if the traffic fingerprint is distinct). This is clearly observable to any modestly capable outside observer. It's even debatable if commonly visited resources are used without any randomized delay between shown alert and the observable event.

> 
> Your second proposition is very valid; it is not clear yet if the application directly shows "you're at risk" or advise to contact health authorities, this will need to be designed with health professionals.
> 
> Would this answer your question ? Thanks

I agree that it has to be designed and evaluated with health professionals in the loop. As the analysis currently stands it's incorrect and incomplete (because you cannot complete the analysis of an incomplete system). It's important that issues like this is highlighted as large number of developers around the world are implementing versions of this proposal.

**Comment by lbarman** [comment]
Absolutely, we agree. This procedure will need to be examined notably with respect to the "eavesdropper" adversary. I think the proposal is voluntarily loose because different countries will have different approaches.

**Comment by pylls** [comment]
Great, thanks. It would then make sense to update the analysis, or at least add some caveats on the design to guide developers for this part.
